backend/test_observer/controllers/execution_metadata/models.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2023 Canonical Ltd.
+#
+# This file is part of Test Observer Backend.
+#
+# Test Observer Backend is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# Test Observer Backend is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+from pydantic import BaseModel, RootModel
+
+from collections import defaultdict
+from collections.abc import Sequence
+
+from test_observer.data_access.models import TestExecutionMetadata
+
+
+class ExecutionMetadata(RootModel[dict[str, list[str]]]):
+    @classmethod
+    def from_rows(cls, values: Sequence[TestExecutionMetadata]) -> "ExecutionMetadata":
```

Comment:
Tried a model_validator for auto conversion of this but mypy was complaining about typing and I couldn't figure that out.